pkg/operator/injection/injection.go

```diff
@@ -57,3 +63,29 @@ func WithOptionsOrDie(ctx context.Context, opts ...options.Injectable) context.C
 	}
 	return ctx
 }
+
+func WithClient(ctx context.Context, client client.Client) context.Context {
+	return context.WithValue(ctx, clientKey, client)
+}
+
+func GetClient(ctx context.Context) client.Client {
+	c := ctx.Value(clientKey)
+	if c == nil {
+		// This is a developer error if this happens, so we should panic
+		panic("client doesn't exist in context")
+	}
+	return c.(client.Client)
+}
+
+func NodeClassToContext(ctx context.Context, opts []schema.GroupVersionKind) context.Context {
```

Comment:
```suggestion
func WithDefaultNodeClass(ctx context.Context, opts []schema.GroupVersionKind) context.Context {
```

This is DefaultNodeClass, right?

Reply:
Can you put a comment over this on why we need this at all

Reply:
No, we want to use the NodeClass schemes as a way to find the APIVersion for going between v1 and v1beta1